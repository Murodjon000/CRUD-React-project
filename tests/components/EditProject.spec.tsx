import shallowSetup from "./mock.spec";
import toJson from "enzyme-to-json";

test("should render EditProject component", () => {
  const { editProjectWrapper, props } = shallowSetup();

  expect(editProjectWrapper.find("div").first().hasClass("form__wrapper")).toBe(
    true
  );
  expect(editProjectWrapper.find("#title").props().value).toBe(
    props.project[0].title
  );
  expect(editProjectWrapper.find("select").props().value).toBe(
    props.project[0].category
  );
  expect(editProjectWrapper.find("option").first().text()).toBe(
    props.project[0].category
  );
  expect(editProjectWrapper.find("#image").props().value).toBe(
    props.project[0].image
  );
  expect(editProjectWrapper.find("#description").props().value).toBe(
    props.project[0].description
  );
});

test("should render EditProject component Snapshot", () => {
  const { editProjectWrapper } = shallowSetup();

  expect(toJson(editProjectWrapper)).toMatchSnapshot();
});
