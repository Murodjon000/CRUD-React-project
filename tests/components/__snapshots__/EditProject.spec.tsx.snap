// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Details component Snapshot 1`] = `
<div
  className="project__details-wrapper"
>
  <div
    className="project__details-wrapper-image"
  >
    <img
      alt=""
      className="test-img"
      src="https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg"
    />
  </div>
  <div
    className="project__details-wrapper-box"
  >
    <h1
      className="test-title"
    >
      Where can I get some?
    </h1>
    <p
      className="test-desc"
    >
       
      ðŸ‘€ PageMaker including versions of Lorem Ipsum.
    </p>
    <button
      className="test-btn"
      onClick={[Function]}
      type="button"
    >
      Back to home
    </button>
  </div>
</div>
`;

exports[`should render EditProject component Snapshot 1`] = `
<div
  className="form__wrapper"
>
  <hr
    className="form__wrapper-line"
  />
  <h1>
    Update Project
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="upadte-form-wrapper"
    >
      <label
        htmlFor="title"
      >
        Title
        <FormHelper
          className="form-update-input"
          id="title"
          onChange={[Function]}
          placeholder="Project title"
          type="text"
          value="Where can I get some?"
        />
      </label>
      <label
        htmlFor="category"
      >
        Select Priority
        <select
          className="form-update-input"
          name="category"
          onChange={[Function]}
          value="High"
        >
          <option
            key="High"
            value="High"
          >
            High
          </option>
          <option
            key="Medium"
            value="Medium"
          >
            Medium
          </option>
          <option
            key="Low"
            value="Low"
          >
            Low
          </option>
        </select>
      </label>
    </div>
    <label
      htmlFor="image"
    >
      Image link
      <FormHelper
        className="form-update-input"
        id="image"
        onChange={[Function]}
        placeholder="Project Image link"
        type="text"
        value="https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg"
      />
    </label>
    <label
      htmlFor="description"
    >
      Description
      <FormHelper
        className="form-update-input"
        id="description"
        onChange={[Function]}
        placeholder="Project description"
        type="textarea"
        value="PageMaker including versions of Lorem Ipsum."
      />
    </label>
    <input
      className="primary-btn"
      type="submit"
      value="Update Project"
    />
  </form>
</div>
`;
